volatility3/framework/layers/crash.py

```diff
@@ -67,34 +72,96 @@ def __init__(self, context: interfaces.context.ContextInterface, config_path: st
+                        last_bit = ((outer_index - 1) * 32) + 31
+                        segment_length = (last_bit - first_bit + 1) * 0x1000
+                        segments.append((first_bit * 0x1000, first_offset, segment_length, segment_length))
+                        first_bit = None
+                elif buffer_long[outer_index] == 0xFFFFFFFF:
+                    if first_bit is None:
+                        first_offset = offset
+                        first_bit = outer_index * 32
+                    offset = offset + (32 * 0x1000)
+                else:
+                    for inner_index in range(0, 32):
+                        bit_addr = outer_index * 32 + inner_index
+                        if (buffer_char[bit_addr >> 3] >> (bit_addr & 0x7)) & 1:
+                            if first_bit is None:
+                                first_offset = offset
+                                first_bit = bit_addr
+                            offset = offset + 0x1000
+                        else:
+                            if first_bit is not None:
+                                segment_length = ((bit_addr - 1) - first_bit + 1) * 0x1000
+                                segments.append((first_bit * 0x1000, first_offset, segment_length, segment_length))
+                                first_bit = None
+                last_bit_seen = (outer_index * 32) + 31
+
+            if first_bit is not None:
+                segment_length = (last_bit_seen - first_bit + 1) * 0x1000
+                segments.append((first_bit * 0x1000, first_offset, segment_length, segment_length))
+        else:
+            vollog.log(constants.LOGLEVEL_VVVV, "unsupported dump format 0x{:x}".format(self.dump_type))
+            raise WindowsCrashDumpFormatException(self.name, "unsupported dump format 0x{:x}".format(self.dump_type))
 
         if len(segments) == 0:
             raise WindowsCrashDumpFormatException(self.name, "No Crash segments defined in {}".format(self._base_layer))
+        else:
+            # report the segments for debugging. this is valuable for dev/troubleshooting but
+            # not important enough for a dedicated plugin.
+            for idx, (start_position, mapped_offset, length, _) in enumerate(segments):
+                vollog.log(constants.LOGLEVEL_VVVV,
+                           "Segment {}: Position {:#x} Offset {:#x} Length {:#x}".format(idx,
+                                                                                         start_position,
+                                                                                         mapped_offset,
+                                                                                         length))
 
         self._segments = segments
 
     @classmethod
     def check_header(cls, base_layer: interfaces.layers.DataLayerInterface, offset: int = 0) -> Tuple[int, int]:
         # Verify the Window's crash dump file magic
-
+        
```

Comment:
Not ideal, but I'm sure it'll get quashed in a future commit.  It suggests yapf hasn't been run over the code though...

Reply:
Fixed in 7d9c66c